tests/unit/replication_persistance_helper.cpp

```diff
@@ -0,0 +1,199 @@
+        "\n"
+        "  ASSERT_TRUE(CheckEquality(replicas_status, *replicas_status_converted));\n"
+        "}}",
+        number_of_tests_failed, name, ip_address, port, sync_mode, timeout, replica_check_frequency, ssl);
+    myfile << test;
+    myfile.close();
+  }
+
+  static_assert(sizeof(memgraph::storage::replication::ReplicaStatus) == 168,
+                "Most likely you modified ReplicaStatus without updating the tests. Please modify CheckEqualityImpl, "
+                "ExportFailingTest and extend the unit tests with relevant tests.");
+};
+
+TEST_F(ReplicationPersistanceHelperTest, BasicTestAllAttributesInitialized) {
+  auto replicas_status = CreateReplicaStatus(
+      "name", "ip_address", 0, memgraph::storage::replication::ReplicationMode::SYNC, 1.0, std::chrono::seconds(1),
+      memgraph::storage::replication::ReplicationClientConfig::SSL{.key_file = "key_file", .cert_file = "cert_file"});
+
+  auto json_status = memgraph::storage::replication::ReplicaStatusToJSON(
+      memgraph::storage::replication::ReplicaStatus(replicas_status));
+  auto replicas_status_converted = memgraph::storage::replication::JSONToReplicaStatus(std::move(json_status));
+
+  ASSERT_TRUE(CheckEquality(replicas_status, *replicas_status_converted));
+}
+
+TEST_F(ReplicationPersistanceHelperTest, BasicTestOnlyMandatoryAttributesInitialized) {
+  auto replicas_status =
+      CreateReplicaStatus("name", "ip_address", 0, memgraph::storage::replication::ReplicationMode::SYNC, std::nullopt,
+                          std::chrono::seconds(1), std::nullopt);
+
+  auto json_status = memgraph::storage::replication::ReplicaStatusToJSON(
+      memgraph::storage::replication::ReplicaStatus(replicas_status));
+  auto replicas_status_converted = memgraph::storage::replication::JSONToReplicaStatus(std::move(json_status));
+
+  ASSERT_TRUE(CheckEquality(replicas_status, *replicas_status_converted));
+}
+
+TEST_F(ReplicationPersistanceHelperTest, RandomizedTests) {
+  auto all_tests_succeeded = true;
+
+  for (auto &name : names_) {
+    for (auto &ip_address : ip_addresses_) {
+      for (auto port : ports_) {
+        for (auto sync_mode : sync_modes_) {
+          for (auto timeout : timeouts_) {
+            for (auto replica_check_frequency : replica_check_frequencies_) {
+              for (auto &ssl : ssls_) {
+                auto replicas_status =
+                    CreateReplicaStatus(name, ip_address, port, sync_mode, timeout, replica_check_frequency, ssl);
+
+                auto json_status = memgraph::storage::replication::ReplicaStatusToJSON(
+                    memgraph::storage::replication::ReplicaStatus(replicas_status));
+                auto replicas_status_converted =
+                    memgraph::storage::replication::JSONToReplicaStatus(std::move(json_status));
+
+                all_tests_succeeded = CheckEquality(replicas_status, *replicas_status_converted) && all_tests_succeeded;
```

Comment:
```suggestion
                ASSERT_TRUE(CheckEquality(replicas_status, *replicas_status_converted));
```

Reply:
Test is removed, no need for that anymore.